src/software/pipeline/main_panoramaWarping.cpp

```diff
@@ -975,11 +979,11 @@ bool computeOptimalPanoramaSize(std::pair<int, int> & optimalSize, const sfmData
 
   
   if (scales.size() > 1) {
```

Comment:
`if (!scales.empty())`
if scales == 1 it should be fine, selected_index will be 0.